lib/components/steps/mapping/MappingTab.tsx

```diff
@@ -248,6 +265,49 @@ const MappingTab = ({defaultCollection}: MappingProps) => {
         return false
     };
 
+    const isSameStrategyType = (filter: CurrentFilterInterface, newFilter: CurrentFilterInterface) => {
+        if (!filter || !newFilter) {
+            return false;
+        }
+        return filter.constructor === newFilter.constructor;
+    }
+
+    const sortRows = useCallback((fitlerStrategy: CurrentFilterInterface) => {
+        const result = fitlerStrategy.doSort(visibleRows, datasetMapping, headerIndexes, getType);
+        setVisibleRows([...result]);
+    }, [datasetMapping, headerIndexes, visibleRows])
+
+    const handleSortOrderChange = useCallback((sortOrder: number) => {
+        if (currentFilterStrategy instanceof StatusFilter) {
+            setStatusSortOrder(sortOrder);
+        } else if (currentFilterStrategy instanceof VariableNameFilter) {
+            setVariableNameSortOrder(sortOrder);
+        } else if (currentFilterStrategy instanceof CdeSortingFilter) {
+            setCdeSortOrder(sortOrder);
+        }
+    }, [currentFilterStrategy])
+
+    const handleSortingStrategy = (newCurrentSortingStrategy: CurrentFilterInterface) => {
+        let newSortOrder = 0;
+        if (currentFilterStrategy && isSameStrategyType(currentFilterStrategy, newCurrentSortingStrategy)) {
+            currentFilterStrategy.toggleSortOrder();
+            newSortOrder = currentFilterStrategy.sortState;
```

Comment:
issue (blocking): My understanding is that we would be using the state variable to control the sortOrder instead of having that responsibility in the currentFilterStrategy

Reply:
This still needs addressing. We either use the sortOrder state variable to manage the order of each filter or we use the filter.sortState. Keeping both seems to cause more troubles (because we need to have them in sync) than good.